File: solutions/cpp/two-fer/1/two_fer.cpp
#include "two_fer.h"
namespace two_fer
{
  auto two_fer() -> string {
    return "One for you, one for me.";
  }
  auto two_fer(const string& name) -> string {
    return string("One for ") + name + ", one for me.";
  }
}

File: solutions/delphi/two-fer/1/TwoFer.pas
unit utwofer;

interface
  function twofer(const name: String = '') : String;

implementation
   function twofer(const name: String = '') : String;
   begin
     if name = '' then
        result := 'One for you, one for me.'
     else
         result := 'One for ' + name + ', one for me.';
   end;
   end.